FirebaseFirestoreUI/FUISnapshotArrayDiff.m

```diff
@@ -377,6 +377,131 @@ - (instancetype)initWithInitialArray:(NSArray<FIRDocumentSnapshot *> *)initial
   return self;
 }
 
+- (void)buildFastDiffs {
```

Comment:
Is it worth a comment to explain the high-level approach and trade-offs? (you mentioned that it may not be a minimal diff, but it's presumably faster).

Also, is the complexity still O(m*n) ? If not, can you update the comment on the FUISnapshotArrayDiff interface?

Reply:
Done.